bluetooth/src/androidLibMain/kotlin/device/DefaultDeviceConnectionManager.kt

```diff
@@ -188,8 +236,8 @@ internal actual class DefaultDeviceConnectionManager(
         val succeeded = when (action) {
             is DeviceAction.Read.Characteristic -> gatt.await().readCharacteristic(action.characteristic.wrapper)
             is DeviceAction.Read.Descriptor -> gatt.await().readDescriptor(action.descriptor.wrapper)
-            is DeviceAction.Write.Characteristic -> writeCharacteristic(action.characteristic, action.newValue)
-            is DeviceAction.Write.Descriptor -> writeDescriptor(action.descriptor, action.newValue)
+            is DeviceAction.Write.Characteristic -> writeCharacteristic(action.characteristic, action.newValue ?: ByteArray(0))
```

Comment:
Where this optional write value comes from (I can't recall why it can be null)? Also writing null or empty byte array make no sense for me (we unwrap it on iOS). On Android this values cannot be null ether (see docs). So fallback into empty ByteArray is not correct here.

My suggestion: make it non-optional in DeviceAction itself, or unwrap on Android (do nothing if null, to align with iOS)